core/arch/interp.c

```diff
@@ -8309,15 +8308,8 @@ fixup_indirect_trace_exit(dcontext_t *dcontext, instrlist_t *trace)
             /* Check for indirect exit. */
             if (is_indirect_branch_lookup_routine(dcontext, (cache_pc)target)) {
                 /* This branch must be a cbnz, or the last_cti was not fixed up. */
-                if (instr->opcode != OP_cbnz) {
-                    prev = instr_get_prev(instr);
-                    added_size += mangle_indirect_branch_in_trace(
```

Comment:
If we remove this, then `mangle_indirect_branch_in_trace` isn't used at all right?

Reply:
It is also called in `fixup_last_cti` on L6537 before we add a block to the trace. In `mixup_indirect_trace_exit`, we should expect it to be already fixed up when we iterate over the trace and check for indirect branch lookup routines, unless something has gone wrong.

Reply:
@abhinav92003 Are you happy with this explanation? If so can you approve this change you have requested?